### PR TITLE
Only load the module specified in run.py argument

### PR DESCRIPTION
**Pull request by xuzhao9**
When running `run.py`, it tries to read and load every model in the `models` directory even though we only need to run one specific model. In practice this often causes problems such as requiring to install `torchvision`/`torchtext` which are not required by the model we are attempting to run.

`run.py` should only load the specific model we are trying to run, instead of loading all the models.

### CODE REVIEW

File: torchbenchmark/__init__.py [hunk 1]
```diff
@@ -464,6 +464,26 @@ def list_models(model_match=None):
                 models.append(Model)
     return models
 
+def load_model_by_name(model):
+    models = filter(lambda x: model.lower() == x.lower(),
+                    map(lambda y: os.path.basename(y), _list_model_paths()))
+    models = list(models)
+    if not models:
+        return None
+    assert len(models) == 1, f"Find more than one models {models} with the exact name: {model}"
```
Comment (aaronenyeshi) [inline torchbenchmark/__init__.py:473]: nit: typo f"Found more than ..."

**Review by aaronenyeshi — APPROVED** [review]
Small nit in comment. Also, does this patch prevent us from matching models with partial name match?

This looks like a good speed up for run.py. Might also be useful for gen_summary_metadata.py.

File: torchbenchmark/__init__.py (hunk 1 above)
Comment (xuzhao9) [reply torchbenchmark/__init__.py]: Thanks for point out!

**Comment by xuzhao9** [comment]
> Small nit in comment. Also, does this patch prevent us from matching models with partial name match?
> 
> This looks like a good speed up for run.py. Might also be useful for gen_summary_metadata.py.

Actually I deliberately avoid partial name matching, because I found it is easy to cause human errors. For example, if user runs "Bert" or "maskrcnn", it may implicitly fallback to a model different from user's intention.

**Review by aaronenyeshi — APPROVED** [review]
> > Small nit in comment. Also, does this patch prevent us from matching models with partial name match?
> > This looks like a good speed up for run.py. Might also be useful for gen_summary_metadata.py.
> 
> Actually I deliberately avoid partial name matching, because I found it is easy to cause human errors. For example, if user runs "Bert" or "maskrcnn", it may implicitly fallback to a model different from user's intention.

I can see where that is coming from. It sounds good to me!

**Pull request closed without merging**